apps/nextjs-example/postcss.config.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ */
+
+module.exports = {
+  plugins: {
+    '@stylexjs/postcss-plugin': {
+      include: [
+        'app/**/*.{js,jsx,ts,tsx}',
+        'components/**/*.{js,jsx,ts,tsx}',
+        // TODO: Figure out a better way to write this
+        './node_modules/@stylexjs/open-props/lib/**/*.{js,jsx,ts,tsx}',
```

Comment:
I tried to use something like this initially
```
require.resolve('@stylexjs/open-props')
```
as suggested [here](https://github.com/facebook/react-strict-dom/pull/238#discussion_r1842825089) but that didn't work because of individual package.json exports. For now I manually specified the files to transform and the example is rendering correctly.

Reply:
We shouldn't ever have to process non `.js` or `.mjs` files from `node_modules` packages, as everything gets compiled to js before publishing.

A more reliable way to get the package exports for any package (inc if they change their exports) is to use a script. Something like this:

```
const fs = require('fs');
const path = require('path');

function getPackageExports(packageName) {
  const packageJsonPath = require.resolve(`${packageName}/package.json`);
  const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));

  if (packageJson.exports) {
    const exports = {};
    for (const key in packageJson.exports) {
      if (typeof packageJson.exports[key] === 'string') {
        exports[key] = require.resolve(path.join(packageName, packageJson.exports[key]));
      } else if (typeof packageJson.exports[key] === 'object') {
        exports[key] = getPackageExports(path.join(packageName, key));
      }
    }
    return exports;
  } else {
    return require.resolve(packageName);
  }
}

const openPropsExports = getPackageExports('@stylexjs/open-props');
```

Reply:
Currently,
```javascript
require.resolve('@stylexjs/open-props/package.json')
```
throws this error:
```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './package.json' is not defined by "exports" in /Users/tsubasa/Projects/github.com/javascripter/stylex/apps/nextjs-example/node_modules/@stylexjs/open-props/package.json
```

[This](https://gist.github.com/javascripter/2d5992e31fa4ed6c08a77a30d3aa3db9) is the postcss.config.js I tried and I found some comments about this:
https://github.com/react-native-community/cli/issues/1168#issuecomment-1192741058
>  Please don't open issues/PRs on open source projects about this. It is not expected that packages export their package.json file. This is a problem React Native needs to fix, not every single package out there.

Is there's a reliable way to find the package root path?

Reply:
I've updated open-props include paths so that it no longer hard-code internal paths (e.g. /lib) in https://github.com/facebook/stylex/pull/781/commits/378c80e5811912511f94a1ad50b85419eb78f687
Not as accurate as I'd like as it doesn't look for exports but it now accounts for node_modules hoisting and should be more stable and explicit about what it does.

Reply:
https://nodejs.org/api/packages.html#package-entry-points

> Existing packages introducing the ["exports"](https://nodejs.org/api/packages.html#exports) field will prevent consumers of the package from using any entry points that are not defined, including the [package.json](https://nodejs.org/api/packages.html#nodejs-packagejson-field-definitions) (e.g. require('your-package/package.json')). This will likely be a breaking change. To make the introduction of ["exports"](https://nodejs.org/api/packages.html#exports) non-breaking, ensure that every previously supported entry point is exported. It is best to explicitly specify entry points so that the package's public API is well-defined.

It's strange that package.json isn't reachable by default, seems like a Node.js issue tbh. But we can export package.json from packages we control